src/precice/impl/Participant.hpp

```diff
@@ -81,6 +82,12 @@ class Participant {
       const mesh::PtrMesh &mesh,
       int                  interpolationOrder);
 
+  /// Adds a configured global \ref Data to the Participant
+  void addGlobalData(
+      const mesh::PtrGlobalData &data,
+      std::string                direction,
+      int                        interpolationOrder = time::Time::DEFAULT_INTERPOLATION_ORDER);
```

Comment:
Not a big fan of the signature of this function. Why not do this identical to `addReadData` and `addWriteData`? You will end up with one more function, but you can spare the errorprone `direction` argument.